test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -1062,6 +1064,43 @@ def test_3q_with_measure(self):
         self.assertTrue(qc_transpiled.size, 1)
         self.assertTrue(qc_transpiled.count_ops().get("measure", 0), 1)
 
+    def test_3q_series(self):
+        """Test a series of 3-qubit blocks."""
+        backend = GenericBackendV2(5, basis_gates=["u", "cx"])
+
+        x = QuantumCircuit(3)
+        x.x(2)
+        x_mat = Operator(x)
+
+        qc = QuantumCircuit(3)
+        qc.unitary(x_mat, range(3))
+        qc.unitary(np.eye(2**3), range(3))
+
+        tqc = transpile(qc, backend, optimization_level=0, initial_layout=[0, 1, 2])
+
+        expected = np.kron(np.eye(2**2), x_mat)
+        self.assertTrue(Operator(tqc).equiv(expected))
+
```

Comment:
Is there a reason that we are checking that the two operators are equivalent vs. being exactly the same?

Reply:
Mainly because I'm not sure if `UnitarySynthesis` is guaranteed to have the correct global phase 🙂 but if it is, then we can use `==` instead